core/fuzzer.h

```diff
@@ -24,37 +31,146 @@ class P4RuntimeFuzzer {
     [[nodiscard]] virtual const ProgramInfo &getProgramInfo() const { return programInfo; }
 
  public:
-    explicit P4RuntimeFuzzer(const ProgramInfo &programInfo) : programInfo(programInfo) {}
+    explicit RuntimeFuzzer(const FuzzerType fuzzerType, const ProgramInfo &programInfo)
+        : fuzzerType(fuzzerType), programInfo(programInfo) {}
+
+    RuntimeFuzzer(const RuntimeFuzzer &) = default;
```

Comment:
I have come around to just removing these autogenerated constructors since the add little value. I would remove them until we actually want to do something with it. I do not think we copy the fuzzer much.